todo_app/app.py

```diff
@@ -1,15 +1,35 @@
-from flask import Flask
-
+from flask import Flask, render_template, request, redirect, url_for
 from todo_app.flask_config import Config
 
+
+from .data import session_items as session
+
 app = Flask(__name__)
 app.config.from_object(Config)
 
 
 @app.route('/')
 def index():
-    return 'Hello World!'
+    #return 'Hello World!'
+    print(session.get_items())
+    return render_template('index.html', items = session.get_items())
+
+
+@app.route('/addItem', methods=['GET', 'POST'])
+def add_item():
+    if request.method == 'POST':
+        item = request.form.get('item') 
+        print(item)
+        return item_added(item)
+    else:
+        return show_add_item_form()
+
+def show_add_item_form():
+        return render_template('itemForm.html')
 
+def item_added(item):
+    session.add_item(item)
+    return redirect(url_for('index'))
```

Comment:
Again, why is this a separate function? I would keep those two lines in the app.route function.

Each function should do one thing (at a certain level of abstraction). You will cover the Single Responsibility Principle briefly in the Module 2 Workshop.

I would say the route function "handles requests to add an item", which covers all of interpreting the request, actually saving the item, and then returning a response.
And you could say `session.add_item` "saves a new item to session storage"
But I don't think this `item_added` function has a clear responsibility.

And that has resulted in a confusing name. "item_added" feels misleading when it's not been added yet. It should maybe be called `add_item`, and maybe you agreed but found it conflicted with the route function's name.